### PR TITLE
Add PrometheusJobFailure alerting rule

### PR DESCRIPTION
**Pull request by marieroque**
Towards https://github.com/giantswarm/giantswarm/issues/23435

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml [hunk 1]
```diff
@@ -52,3 +52,16 @@ spec:
         team: atlas
         topic: observability
         cancel_if_outside_working_hours: "true"
+    - alert: PrometheusJobFailures
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:55]:
```suggestion
    - alert: PrometheusJobScrapingFailure
```

Is that name not more meaningful?

Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:55]: Maybe, but I also dislike longer names. Just a preference.

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:55]: Why do you dislike longer names? Being able to know at a glance what this alert is makes everything easier

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml:55]: As an example: 
PrometheusJobFailures hints that a job (i.e. cronjob) named Prometheus fails to run or is failing, having the notion of scraping ensure we knowwhat this is about :)

**Review by TheoBrigitte — APPROVED** [review]

**Pull request merged**